### PR TITLE
typescript sdk supports every option supported for the backend for fe…

### PR DESCRIPTION
**Pull request by DanielCollins**
…tching

a single entity

### CODE REVIEW

File: typescript/src/entities/product.test.ts [hunk 1]
```diff
@@ -29,6 +29,18 @@ test('can fetch product from server', () => {
...
 });
 
+test('can specify options in request', () => {
+  const merchi = new Merchi();
+  const testName = 'S7qHUfV_dr5l';
+  mockFetch(true, {'product': {'name': testName}}, 200);
+  (window as any).merchiBackendUri = 'http://override.example.com/';
+  const options = {includeArchived: true,
+                   withRights: true}; 
+  return merchi.Product.get(1, options).then(product => expect(product.name).toBe(testName));
```
Comment (s3341458) [inline typescript/src/entities/product.test.ts]: this testcase can only ensure when you add  "includeArchived", "withRights" to options does not break get function.

Please write a testcase for backendFetch ensure it fetch(url.toString(), options) been called with correct parameters.

Comment (s3341458) [reply typescript/src/entities/product.test.ts]: my experience of jest is very limited but suppose able to do it by
>
// The first arg of the first call to the function was 'first arg'
expect(tetch.mock.calls[0][0]).toBe('first arg');

Comment (DanielCollins) [reply typescript/src/entities/product.test.ts]: yes

File: typescript/src/test_util.ts [hunk 2]
```diff
@@ -5,11 +5,14 @@ export function setup() {
 }
 
 export function mockFetch(ok: boolean, data: any, status: number) {
-  (global as any).fetch = jest.fn().mockImplementation(() =>
+  const mock = jest.fn();
```
Comment (s3341458) [inline typescript/src/test_util.ts]: I personally favors the old way you do it, but I do not have strong personal opinion about it

File: typescript/src/entities/product.test.ts [hunk 3]
```diff
@@ -30,6 +30,19 @@ test('can fetch product from server', () => {
   return merchi.Product.get(1).then(product => expect(product.name).toBe(testName));
 });
 
+test('can specify options in request', () => {
+  const merchi = new Merchi();
+  const testName = 'S7qHUfV_dr5l';
+  const fetch = mockFetch(true, {'product': {'name': testName}}, 200);
+  (window as any).merchiBackendUri = 'http://override.example.com/';
```
Comment (s3341458) [inline typescript/src/entities/product.test.ts]: due to the merge of https://github.com/merchisdk/sdk/pull/198 

this line is not valid anymore. 

I will approve it, but I suggest you fix it here before you merge it

**Pull request merged**